src/test-unit/Redis/Locking/RedisDistributedAppLockTest.cs

```diff
@@ -179,5 +179,15 @@ public async Task Properly_traps_other_exceptions()
             Assert.Equal(DistributedAppLockExceptionReason.SeeInnerException, ex.Reason);
             Assert.IsType<OperationCanceledException>(ex.InnerException);
         }
+        
+        [Fact]
+        public async Task Does_set_lock_take_time_and_happened_immediately()
+        {
+            var handle = new RedisDistributedAppLock(_manager, _rng);
+            await handle.AcquireLockAsync(_defaultLockName);
+            
+            Assert.True(handle.WasAcquiredInstantly);
+            Assert.True(handle.TimeUsedToAcquire.TotalMilliseconds > 0);
```

Comment:
**suggestion**: Asserting on `Ticks` might be more robust, as we could get lucky with a test that manages to connect faster than 1 ms thereby creating a false negative.